lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -482,6 +482,10 @@ public function diffColumn(Column $column1, Column $column2)
             if ($properties1['fixed'] !== $properties2['fixed']) {
                 $changedProperties[] = 'fixed';
             }
+        } elseif ($properties1['type'] instanceof Types\BlobType) {
+            if ($properties1['length'] !== $properties2['length']) {
```

Comment:
Looks like it will introduce a false-positive diff when comparing say `BLOB(1)` and `BLOB(2)` which both would translate to a `TINYBLOB` in MySQL.

While the fix seems to be within the current comparator design, the design obviously has flaws. A proper approach might be to add platform awareness to the comparator API and compare the SQL generated against a given platform.

Reply:
Agreed, but changing the comparator API cannot be done in a bugfix release I guess?

Reply:
`2.12.2` and `2.13.0` should be released at about the same time, so the release schedule shouldn't be a concern. If it's a non-breaking change (e.g. that adds a new optional parameter to the API), it could be released in `2.13.0`. Why does it have to be a bugfix release?

Reply:
> Why does it have to be a bugfix release?

I would consider it a bug because the MySQL diff for blob columns doesn’t work currently.

> Looks like it will introduce a false-positive diff when comparing say `BLOB(1)` and `BLOB(2)`

Wouldn’t such false positives also occur for many other types already? Like `binary(fixed=true)` vs `binary(fixed=false)` on PostgreSQL/SQLite or `smallint` vs `integer` on SQLite?